### PR TITLE
Improvement: Limit scale value in 3DP to avoid object dimension setting to 0

### PR DESCRIPTION
**Pull request by jane-rose**

### CODE REVIEW

File: src/app/widgets/PrintingVisualizer/VisualizerModelTransformation.jsx [hunk 1]
```diff
@@ -286,7 +290,7 @@ class VisualizerModelTransformation extends PureComponent {
                             <span className={classNames(styles['axis-label'], styles['axis-red'])}>X</span>
                             <span className={styles['axis-input-1']}>
                                 <Input
-                                    min={0}
+                                    min={1}
```
Comment (parachvte) [inline src/app/widgets/PrintingVisualizer/VisualizerModelTransformation.jsx:293]: 1 or 0.01?

File: src/app/components/SMCanvas/TransformControls.js [hunk 2]
```diff
@@ -841,6 +845,19 @@ class TransformControls extends Object3D {
...
     }
 
+    limitScaleValue(parentEVec) {
+        let shouldDisableScale = false;
+        this.object.children.forEach((meshObject) => {
+            if (parentEVec.x * meshObject.scale.x < 0.01
+                || parentEVec.y * meshObject.scale.y < 0.01
+                || parentEVec.z * meshObject.scale.z < 0.01
+            ) {
+                shouldDisableScale = true;
```
Comment (parachvte) [inline src/app/components/SMCanvas/TransformControls.js:855]: break or return

File: src/app/components/SMCanvas/TransformControls.js [hunk 3]
```diff
@@ -841,6 +845,19 @@ class TransformControls extends Object3D {
         this.dispatchEvent(EVENTS.UPDATE);
     }
 
+    limitScaleValue(parentEVec) {
```
Comment (parachvte) [inline src/app/components/SMCanvas/TransformControls.js:848]: bad naming

File: src/app/components/SMCanvas/TransformControls.js [hunk 4]
```diff
@@ -819,8 +821,10 @@ class TransformControls extends Object3D {
                     parentEVec.y = (this.axis === 'Y' ? parentEVec.y : 1);
                     parentEVec.z = (this.axis === 'Z' ? parentEVec.z : 1);
                 }
-
-                this.object.scale.copy(this.scaleStart).multiply(parentEVec);
+                shouldDisableScale = this.limitScaleValue(parentEVec);
```
Comment (parachvte) [inline src/app/components/SMCanvas/TransformControls.js:824]: variable `shouldDisableScale` is not needed.

File: src/app/widgets/PrintingVisualizer/VisualizerModelTransformation.jsx (hunk 1 above)
Comment (jane-rose) [reply src/app/widgets/PrintingVisualizer/VisualizerModelTransformation.jsx:293]: 1

File: src/app/components/SMCanvas/TransformControls.js (hunk 2 above)
Comment (jane-rose) [reply src/app/components/SMCanvas/TransformControls.js:855]: inside 'forEach' can't use break or return, i change it to 'every'

Comment (parachvte) [reply src/app/components/SMCanvas/TransformControls.js:853]: You can return the value of `Array.every()`

```JavaScript
shouldApplyScaleToObjects(parentEVec) {
  return this.object.children.every((meshObject) => {
    if (condition) {
      return false; // should disable 
    }
    return true;
  });
}
```

**Comment by parachvte** [comment]
LGTM

**Pull request merged**